packages/eslint-plugin/typings/ts-eslint.d.ts

```diff
@@ -397,8 +397,159 @@ declare module 'ts-eslint' {
 
   // This isn't the correct signature, but it makes it easier to do custom unions within reusable listneers
   // never will break someone's code unless they specifically type the function argument
-  type RuleFunction = (node: never) => void;
-  type RuleListener = Record<string, RuleFunction>;
+  type RuleFunction<T = never> = (node: T) => void;
```

Comment:
what do you think about changing this to
```suggestion
  type RuleFunction<T extends TSESTree.Node = never> = (node: T) => void;
```

Reply:
`Comment` does not conform to `Node`. Had to extend `BaseNode`. Please take a look.